core/src/main/java/org/opensearch/sql/planner/optimizer/LogicalPlanOptimizer.java

```diff
@@ -52,11 +51,11 @@ public static LogicalPlanOptimizer create() {
          * Phase 2: Transformations that rely on data source push down capability
          */
         new CreateTableScanBuilder(),
-        new CreatePagingTableScanBuilder(),
         TableScanPushDown.PUSH_DOWN_FILTER,
         TableScanPushDown.PUSH_DOWN_AGGREGATION,
         TableScanPushDown.PUSH_DOWN_SORT,
         TableScanPushDown.PUSH_DOWN_LIMIT,
+        new PushDownPageSize(),
```

Comment:
Does it really follow the order of rules?

Reply:
Optimizer visits nodes in the query plan in depth-first manner -- table scan is visited first, then the plan that has the table scan as a child, and so on.  For each node, the rules are checked and applied in the oder listed here.

If two rules apply to the same node and the order matters, then this list determines it.